test/message-generator/mock/jds-mock-quick-templates.json

```diff
@@ -0,0 +1,172 @@
+        },
+        {
+            "type": "automatic",
+            "message_id": "declare_mining_job_success_1"
+        },
+        {
+            "type": "automatic",
+            "message_id": "test/message-generator/messages/common_messages.json::setup_connection_success_tproxy"
+        },
+        {
+            "type": "automatic",
+            "message_id": "test/message-generator/messages/common_messages.json::setup_connection_success_jds_to_jdc"
+        }
+    ],
+    "actions": [
+       {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "match_message_type",
+                    "value": "0x00"
+                }
+            ],
+            "actiondoc": "This action checks that a SetupConnection message is received,"
+        },
+        {
+            "message_ids": ["setup_connection_success_jds_to_jdc"],
+            "role": "server",
+            "results": [],
+            "actiondoc": "This action sends back a SetupConnection.Success (setup_connection_success_jds_to_jdc) "
+        },
+        {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "get_message_field",
+                    "value": [
+                        "JobDeclarationProtocol",
+                        "AllocateMiningJobToken",
+                        [
+                            [
+                                "request_id",
+                                "request_id_0"
+                            ]
+                        ]
+                    ]
+                }
+            ],
+            "actiondoc": "This action checks receives allocate mining job"
+        },
+        {
+            "message_ids": ["allocate_mining_job_token_success_0"],
+            "role": "server",
+            "results": [],
+            "actiondoc": "This action sends back a AllocateMiningJobToken.Success (allocate_mining_job_token_success_0)"
+        },
+        {
+            "message_ids": [],
+            "role": "server",
+            "results": [
+                {
+                    "type": "get_message_field",
+                    "value": [
+                        "JobDeclarationProtocol",
+                        "AllocateMiningJobToken",
+                        [
+                            [
+                                "request_id",
+                                "request_id_1"
+                            ]
+                        ]
+                    ]
+                }
+            ],
+            "actiondoc": "This action receives allocate mining job"
+        },
+        {
+            "message_ids": ["allocate_mining_job_token_success_1"],
+            "role": "server",
+            "results": [],
+            "actiondoc": "This action sends back a AllocateMiningJobToken.Success (allocate_mining_job_token_success_1)"
+        }
```

Comment:
I see there's still no action for receiving a `DeclareMiningJob`

you probably declared it locally, but you noticed the message wasn't arriving and you never committed it

please see the comments on TP mock, where I explain a possible root cause

Reply:
Exactly, I removed it manually... I will revert back once the TP stuff resolves..